index.es6.js

```diff
@@ -0,0 +1,305 @@
+
+    return axios(params)
+      .then(function (response) {
+        return response.data;
+      })
+      .catch(function (error) {
+        throw error.data;
+      });
+  }
+
+  asset (id, callback) {
+    return nodeify(
+      this._request('/assets/' + id).then(parseResource),
+      callback
+    );
+  }
+
+  assets (object, callback) {
+    var query = Query.parse(object);
+    var deferred = this._request('/assets', query)
+                   .then(parseSearchResult);
+    return nodeify(deferred, callback);
+  }
+
+  contentType (id, callback) {
+    var deferred = this._request('/content_types/' + id)
+                   .then(ContentType.parse);
+    return nodeify(deferred, callback);
+  }
+
+  contentTypes (object, callback) {
+    var query = Query.parse(object);
+    var deferred = this._request('/content_types', query)
+                   .then(parseSearchResult);
+    return nodeify(deferred, callback);
+  }
+
+  entry (id, callback) {
+    var deferred = this._request('/entries/' + id)
+                   .then(Entry.parse);
+    return nodeify(deferred, callback);
+  }
+
+  entries (object, callback) {
+    var query = Query.parse(object);
+    var deferred = this._request('/entries', query)
+                   .then(parseSearchResult);
+    return nodeify(deferred, callback);
+  }
+
+  space (callback) {
+    return nodeify(this._request(''), callback);
+  }
+
+  _pagedSync (sync) {
+    var self = this;
+    return this._request('/sync', sync.query)
+               .then(function (data) {
+                 sync.append(data);
+                 if(!sync.done){
+                   return self._pagedSync(sync);
+                 } else {
+                   return {
+                     items: sync.items,
+                     nextSyncToken: sync.nextSyncToken
+                   };
+                 }
+               });
+  }
+
+  sync (object, callback) {
+    if (!object || (!object.initial && !object.nextSyncToken)) {
+      throw new Error('Please provide either the initial flag or a nextSyncToken for syncing');
+    }
+    var query = Query.parse(object);
+    var deferred = this._pagedSync(new Sync(query))
+                   .then(function (response) {
+                     response.items = parseSearchResult(response);
+                     return response;
+                   });
+    return nodeify(deferred, callback);
+  }
+}
+
+class Asset {
+  static parse (object) {
+    return _.extend(new Asset(), {
+      sys: Sys.parse(object.sys),
+      fields: object.fields
+    });
+  }
+}
+
+class Entry {
+  static parse (object) {
+    return _.extend(new Entry(), {
+      sys: Sys.parse(object.sys),
+      fields: object.fields
+    });
+  }
+}
+
+class ContentType {
+  static parse (object) {
+    return _.extend(new ContentType(), {
+      sys: Sys.parse(object.sys),
+      fields: object.fields.map(Field.parse),
+    }, _.pick(object, 'name', 'displayField'));
```

Comment:
[Destructuring](https://babeljs.io/docs/learn-es6/#modules) might make this more readable
~~~js
static parse ({name, sys, fields, displayField}) {
  return _.extend(new ContentType(), {
    name: name,
    sys: Sys.parse(sys),
    fields: fields.map(Field.parse),
    displayField: displayField
  })
}
~~~